plugins/content/bower/index.js

```diff
@@ -837,7 +837,7 @@ function handleUploadedPlugin (req, res, next) {
 
         // first entry should be our target directory
         var packageJson;
-        var canonicalDir = path.join(outputPath, files[0]);
+        var canonicalDir = path.join(outputPath, file.name.replace('.zip', ''));
```

Comment:
maybe the solution is to iterate the directories and search for the existence of the bower.json file?

````
// pseudocode
var packagePath = false;
for each file in files {
  var bowerTest = path.join(outputPath, file.name, 'bower.json');
  if (exists(bowerTest)) {
    // found our man
    packagePath = bowerTest;
    break;
  }
}

if (packagePath) {
  // do stuff with packagePath
}
````